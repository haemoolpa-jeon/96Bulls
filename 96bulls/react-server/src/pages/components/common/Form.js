import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styles from './Form.module.scss';

class Form extends Component {
  handleSubmit = (e) => {
    const { onSubmit } = this.props;
    onSubmit(e);
  };

  render() {
    const { children, className } = this.props;
    return (
      <form className={`${styles.form} ${className}`} onSubmit={this.handleSubmit}>
        {children}
      </form>
    );
  }
}

Form.propTypes = {
  children: PropTypes.node.isRequired,
  onSubmit: PropTypes.func,
  className: PropTypes.string,
};

export default Form;
