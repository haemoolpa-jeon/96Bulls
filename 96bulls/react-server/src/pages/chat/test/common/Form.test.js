import React from 'react';
import Form from '../../components/common/Form';
import {shallow} from 'enzyme/build';

describe('Form', () => {
  let component = null;
  const mockCallback = jest.fn(e => e);
  it('초기 렌더링이 문제없이 되야함', () => {
    component = shallow(<Form
      onSubmit={mockCallback}>
      <input type='text'/>
    </Form>);
  });

  it('초기 렌더링 스냅샷 일치함', () => {
    expect(component).toMatchSnapshot();
  });

  it('onSubmit 함수가 수행되어야 함', () => {
    component.find('form').simulate('submit');
    expect(mockCallback.mock.calls.length).toBe(1);
  });
});
