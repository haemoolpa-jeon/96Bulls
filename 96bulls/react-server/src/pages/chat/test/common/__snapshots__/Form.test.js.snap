// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form 초기 렌더링 스냅샷 일치함 1`] = `
<form
  className="form undefined"
  onSubmit={[Function]}
>
  <input
    type="text"
  />
</form>
`;
